onadata/apps/api/permissions.py

```diff
@@ -277,6 +277,9 @@ def has_object_permission(self, request, view, obj):
                 isinstance(obj.content_object, DataView)):
             return False
 
+        if 'key' in request.query_params or view.action == 'partial_update':
```

Comment:
No it is not. I am looking for a way to find the user's permission in the parent Xform and restrict that in the creation and changing of the widget.